### PR TITLE
Clock synchronization issues

### PR DESCRIPTION
**Pull request by ramonsmits**
Based on issue https://github.com/Particular/PlatformDevelopment/issues/978 and the [comment from @tmasternak](https://github.com/Particular/PlatformDevelopment/issues/978#issuecomment-253770944)

### CODE REVIEW

File: nservicebus/messaging/discard-old-messages.md [hunk 1]
```diff
@@ -38,6 +38,18 @@ snippet:DiscardingOldMessagesWithCode
...
 
+## Clock synchronization issues
+
+When sending a message with a certain TimeToBeReceived value it could happen that the receiver drops the message if clocks are too much out of sync. For example, if TimeToBeReceived is 1 minute and the receivers clock is 1 minute ahead compared to the sender the message could potentially never be delivered thus processed.
+
+Because clocks usually are at most a few minutes out of sync this issue only applies to relatively small TimeToBeReceived values.
+
+For this reason it is wise to add the maximum amount of allowed clock offset, called clock drift, to the TTBR value,
+
+For example, you want to use a TimeToBeReceived value of 90 seconds, we allow for 300 seconds of maximum clock drift so the TTBR value becomes 90 + 300 = 390 seconds.
```
Comment (SimonCropp) [inline nservicebus/messaging/discard-old-messages.md]: remove the personal voice

**Comment by weralabaj** [comment]
@ramonsmits is the value of 1 min. arbitrary? should we mention how they can determine it? I'm ok to keep as is, if that is too complicated.

I fixed the personal voice issue, so ready to merge if you're ok.

**Comment by ramonsmits** [comment]
@weralabaj 

> is the value of 1 min. arbitrary?

The 1 minute is the TTBR value that the customer wants to apply.

File: nservicebus/messaging/discard-old-messages.md [hunk 2]
```diff
@@ -38,6 +38,15 @@ snippet:DiscardingOldMessagesWithCode
 TimeToBeReceived relies on underlying functionality in the transport infrastructure to discard expired messages. This feature's usefulness is highly affected by the actual implementation in the different transports.
 
 
+## Clock synchronization issues
+
+When sending a message with a certain TimeToBeReceived value it could happen that the receiver drops the message if clocks are too much out of sync. For example, if TimeToBeReceived is 1 minute and the receivers clock is 1 minute ahead compared to the sender the message could potentially never be delivered thus processed.
+
+Because clocks usually are at most a few minutes out of sync this issue only applies to relatively small TimeToBeReceived values.
+
+For this reason it is wise to add the maximum amount of allowed clock offset, called clock drift, to the TTBR value. For example, when using TimeToBeReceived value of 90 seconds, one should allow for 300 seconds of maximum clock drift so the TTBR value becomes 90 + 300 = 390 seconds.
```
Comment (ramonsmits) [inline nservicebus/messaging/discard-old-messages.md]: @weralabaj 

> one should allow for 300 seconds of maximum clock drift

I don't know for sure but I don't think that is correct. I read *one should allow* meaning that in this case a user should allow for this amount of clock drift.

This is my original sentence:

> For example, you want to use a TimeToBeReceived value of 90 seconds, we allow for 300 seconds of maximum clock drift so the TTBR value becomes 90 + 300 = 390 seconds.

The 300 seconds is a value that should be appropriate to the environment of the customer.

**Pull request merged**